hugolib/config.go

```diff
@@ -283,8 +283,14 @@ func (l configLoader) loadConfigFromConfigDir(v *viper.Viper) ([]string, error)
 				return nil
 			}
 
-			name := helpers.Filename(filepath.Base(path))
+			for _, filenamePart := range strings.Split(fi.Name(), ".") {
+				// Ignore .swp files
+				if filenamePart == "swp" {
```

Comment:
I don't use vim, but should that check be case insensitive?

Reply:
Editing a file with Vim generates a `.swp` file. It works as a [buffer](https://vi.stackexchange.com/questions/177/what-is-the-purpose-of-swap-files).
That check does not need to be case sensitive since the generation of these files is automatic.

But your comment showed me that something is incorrect here: the `swp` is the file extension, we only need the test with the last part of the filename. Also, the .swp files are hidden files.

Let me correct it.

Reply:
I understand the problem, but the fix is sub-optimal.

* We have a list of config file extensions we support (lower case, see another PR that pulls this list into a variable)
* Every file that is not in this list can be safely ignored

Reply:
Here is my config folder:
```sh
config
├── _default
│   ├── config.toml
│   ├── .config.toml.swp # while editing config.tom with Vim
```

Here is the error message: 
```sh
Error: "/Users/fberrez/workspace/src/github.com/fberrez/hugotest/config/_default/.config.toml.swp:1:1": "/Users/fberrez/workspace/src/github.com/fberrez/hugotest/config/_default/.config.toml.swp" is not a valid configuration format
```

According to your answer, the `validConfigFilenames` slice containing all valid config file extensions would ignore the `swp` file. But, it returns an error.

Therefore, we have to change the checker behavior, no?

Reply:
>Therefore, we have to change the checker behavior, no?

 Yes, you have to change. But there is a fundamental difference between "if ext == swp then continue" an "if ext not in valid list then continue"